src/components/flat-table/flat-table-row/flat-table-row.component.js

```diff
@@ -16,28 +40,62 @@ const FlatTableRow = React.forwardRef(
       }
     }
 
-    if (onClick) {
+    function handleClick(ev) {
+      if (onClick) onClick(ev);
+      if (expandable && expandableArea === "wholeRow") {
+        setIsExpanded(!isExpanded);
+      }
+    }
+
+    if (onClick || expandable) {
       interactiveRowProps = {
         isRowInteractive: true,
         tabIndex: 0,
-        onClick,
         onKeyDown,
+        isFirstColumnInteractive: expandableArea === "firstColumn",
+        isExpanded,
       };
     }
 
     return (
       <SidebarContext.Consumer>
         {(context) => (
-          <StyledFlatTableRow
-            isInSidebar={context && context.isInSidebar}
-            data-element="flat-table-row"
-            highlighted={highlighted}
-            selected={selected}
-            ref={ref}
-            {...interactiveRowProps}
-          >
-            {children}
-          </StyledFlatTableRow>
+          <>
+            <StyledFlatTableRow
+              isInSidebar={context && context.isInSidebar}
+              expandable={expandable}
+              isSubRow={isSubRow}
+              isFirstSubRow={isFirstSubRow}
+              data-element={isSubRow ? "flat-table-sub-row" : "flat-table-row"}
+              highlighted={highlighted}
+              selected={selected}
+              onClick={handleClick}
+              ref={ref}
+              {...interactiveRowProps}
+            >
+              {React.Children.map(children, (child, index) => {
+                return React.cloneElement(child, {
```

Comment:
We might have to do some optimisations down the line as I think it will clone each child every time a row or sub row is clicked for example. I don't think we can avoid cloning though so just commenting rather than providing an alternative 😆